src/modules/rust.rs

```diff
@@ -19,27 +23,85 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         return None;
     }
 
-    match get_rust_version() {
-        Some(rust_version) => {
-            const RUST_CHAR: &str = "🦀 ";
+    let module_version = if let Some(toolchain) = env_rustup_toolchain()
+        .or_else(|| execute_rustup_override_list(&context.current_dir))
+        .or_else(|| find_rust_toolchain_file(&context.current_dir))
+    {
+        match execute_rustup_run_rustc_version(&toolchain) {
+            Err(err) => err,
+            Ok(stdout) => format_rustc_version(stdout),
+        }
+    } else {
+        format_rustc_version(execute_rustc_version()?)
+    };
+
+    let mut module = context.new_module("rust");
 
-            let mut module = context.new_module("rust");
-            let module_style = module
-                .config_value_style("style")
-                .unwrap_or_else(|| Color::Red.bold());
-            module.set_style(module_style);
+    let module_style = module
+        .config_value_style("style")
+        .unwrap_or_else(|| Color::Red.bold());
+    module.set_style(module_style);
+    module.new_segment("symbol", RUST_CHAR);
+    module.new_segment("version", &module_version);
 
-            let formatted_version = format_rustc_version(rust_version);
-            module.new_segment("symbol", RUST_CHAR);
-            module.new_segment("version", &formatted_version);
+    Some(module)
+}
 
-            Some(module)
+fn env_rustup_toolchain() -> Option<String> {
+    let val = env::var("RUSTUP_TOOLCHAIN").ok()?;
+    Some(val.trim().to_owned())
+}
+
+fn execute_rustup_override_list(cwd: &Path) -> Option<String> {
+    let Output { stdout, .. } = Command::new("rustup")
+        .args(&["override", "list"])
+        .output()
+        .ok()?;
+    let stdout = String::from_utf8(stdout).ok()?;
+    if stdout == "no overrides\n" {
+        return None;
+    }
+    stdout
```

Comment:
Would be great to move this `stdout` parsing logic in to another function, where we could write a test or two against it?

Reply:
Right. I moved the part to `extract_toolchain_from_rustup_override_list` and added a test for it.